backend/app/api/v1/commons/telco.py

```diff
@@ -40,58 +45,71 @@ async def getData(
-                "ciSystem": "Jenkins",
-                "benchmark": test_data["test_type"],
-                "kernel": kernel,
-                "shortVersion": test_data["ocp_version"],
-                "ocpVersion": test_data["ocp_build"],
-                "releaseStream": utils.getReleaseStream(
-                    {"releaseStream": test_data["ocp_build"]}
-                ),
-                "nodeName": test_data["node_name"],
-                "cpu": test_data["cpu"],
-                "formal": test_data["formal"],
-                "startDate": str(start_time_utc),
-                "endDate": str(end_time_utc),
-                "buildUrl": jenkins_url
-                + "/"
-                + str(test_data["cluster_artifacts"]["ref"]["jenkins_build"]),
-                "jobStatus": "failure" if (threshold != 0) else "success",
-                "jobDuration": execution_time_seconds,
-            }
-        )
+    if response:
+        for each_response in response["data"]:
+            end_timestamp = int(each_response["timestamp"])
+            test_data = each_response["data"]
+            threshold = await telcoGraphs.process_json(test_data, True)
+            hash_digest, encrypted_data = hasher.hash_encrypt_json(each_response)
+            execution_time_seconds = test_type_execution_times.get(
+                test_data["test_type"], 0
+            )
+            start_timestamp = end_timestamp - execution_time_seconds
+            start_time_utc = datetime.fromtimestamp(start_timestamp, tz=timezone.utc)
+            end_time_utc = datetime.fromtimestamp(end_timestamp, tz=timezone.utc)
+            kernel = test_data["kernel"] if "kernel" in test_data else "Undefined"
+
+            mapped_list.append(
+                {
+                    "uuid": hash_digest,
+                    "encryptedData": encrypted_data.decode("utf-8"),
+                    "ciSystem": "Jenkins",
+                    "benchmark": test_data["test_type"],
+                    "kernel": kernel,
+                    "shortVersion": test_data["ocp_version"],
+                    "ocpVersion": test_data["ocp_build"],
+                    "releaseStream": utils.getReleaseStream(
+                        {"releaseStream": test_data["ocp_build"]}
+                    ),
+                    "nodeName": test_data["node_name"],
+                    "cpu": test_data["cpu"],
+                    "formal": test_data["formal"],
+                    "startDate": str(start_time_utc),
+                    "endDate": str(end_time_utc),
+                    "buildUrl": jenkins_url
+                    + "/"
+                    + str(test_data["cluster_artifacts"]["ref"]["jenkins_build"]),
+                    "jobStatus": "failure" if (threshold != 0) else "success",
+                    "jobDuration": execution_time_seconds,
+                }
+            )
 
     jobs = pd.json_normalize(mapped_list)
 
     return {"data": jobs, "total": response["total"]}
```

Comment:
You added an `if response` check, but you then use the response value afterwards, not conditioned by it. What situation does this happen, and why doesn't it also apply to the "total" key/value?
If you want to do something else in that situation, you can add a condition for the not response, and exit early.

Reply:
Added the check for `total` too!